### PR TITLE
Create-Battle-Endpoint-Test

### PR DESCRIPTION
**Pull request by eduardavercosa**
## Objective and Motivation
- Test if Create-Battle-Endpoint works

### CODE REVIEW

File: backend/battling/tests/test_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        self.opponent2 = baker.make("users.User")
+
+    def test_cannot_create_battle_with_user_not_logged(self):
+        self.auth_client.logout()
+        battle_data = {
+            "creator_id": self.user.id,
+            "opponent_id": self.opponent.id,
+        }
+
+        self.auth_client.post(reverse("create-battle"), battle_data)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: Please assert that the response is 403

File: backend/battling/tests/test_endpoints.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            "ERROR: You can't challenge yourself.",
+        )
+
+    def test_create_battle_successfully(self):
+        battle_data = {
+            "creator_id": self.user.id,
+            "opponent_id": self.opponent.id,
+        }
+
+        self.auth_client.post(reverse("create-battle"), battle_data)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: Assert that the response status is correct

File: backend/battling/tests/test_endpoints.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+class BattleListEndpointTest(TestCaseUtils):
+    def setUp(self):
+        super().setUp()
+        self.user2 = baker.make("users.User")
+
+    # def test_user_cannot_acess_battle_list_logged_out(self):
+    #     self.auth_client.logout()
+    #     response = self.auth_client.get(reverse('battle-list'))
+    #     import ipdb; ipdb.set_trace()
+    #     self.assertEqual(response.status_code, 200)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: Why is this commented?

File: backend/battling/tests/test_endpoints.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    def test_no_battles_in_list(self):
+        response = self.auth_client.get(reverse("battle-list"))
+        self.assertEqual(response.json(), [])
+
+    def test_one_battle_in_list(self):
+        battle = baker.make("battling.Battle", creator=self.user)
+        self.auth_client.get(reverse("battle-list"))
+        BattleSerializer([battle], many=True)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: this is not being used, but please check if the contents are the same

File: backend/battling/tests/test_endpoints.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+        battles = Battle.objects.filter(creator=self.user)
+        assert len(battles) == 1
+
+    def test_a_few_battles_in_list(self):
+        battle = baker.make("battling.Battle", creator=self.user, _quantity=5)
+        self.auth_client.get(reverse("battle-list"))
+        BattleSerializer([battle], many=True)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: same as before

File: backend/battling/tests/test_endpoints.py (hunk 4 above)
Comment (rsarai) [reply backend/battling/tests/test_endpoints.py]:
```
response = self.auth_client.get(reverse("battle-list"))
expected_response = BattleSerializer([battle], many=True)
self.assertCountEqual(expected_response.data, response.json())
```

File: backend/battling/tests/test_endpoints.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+        battles = Battle.objects.filter(creator=self.user)
+        assert len(battles) == 5
+
+    def test_a_lot_battle_in_list(self):
+        battle = baker.make("battling.Battle", creator=self.user, _quantity=50)
+        self.auth_client.get(reverse("battle-list"))
+        BattleSerializer([battle], many=True)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: same as before

File: backend/battling/tests/test_endpoints.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+        battles = Battle.objects.filter(creator=self.user)
+        assert len(battles) == 50
+
+    def test_battle_with_multiple_requests_in_list(self):
+        battle = baker.make("battling.Battle", creator=self.user, _quantity=10)
+        battle2 = baker.make("battling.Battle", creator=self.user2, _quantity=9)
+        self.auth_client.get(reverse("battle-list"))
+        BattleSerializer([battle], many=True)
+        BattleSerializer([battle2], many=True)
```
Comment (rsarai) [inline backend/battling/tests/test_endpoints.py]: same as before, this is not being used

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/templates/registration/invite_signup_email.html [hunk 8]
```diff
@@ -3,7 +3,7 @@
 
 {% translate "Please go to the following page and register:" %}
 {% block reset_link %}
-{{ HOST }}{% url 'signup_confirm' uidb64=uid token=token %}
+{{ domain }}{% url 'signup_confirm' uidb64=uid token=token %}
```
Comment (rsarai) [inline backend/templates/registration/invite_signup_email.html]: You also need to add the `{{ protocol }}://` no? Or is this already on the domain?

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/templates/registration/invite_signup_email.html (hunk 8 above)
Comment (eduardavercosa) [reply backend/templates/registration/invite_signup_email.html]: It's already in the domain :D

**Review by rsarai — APPROVED** [review]

**Pull request merged**